content/core-features/authentication.textile

```diff
@@ -373,6 +374,127 @@ The following JavaScript example demonstrates how to issue an "Ably Token":#ably
+
+h4. API endpoint
+
+The Ably REST API contains the following endpoint for token revocation:
+
+```
+POST /keys/<keyName>/revokeTokens
+```
+
+Where the @keyName@ is @appId.keyId@.
+
+In that request, the request body has the following form:
+
+```[json]
+{
+    targets: [/* ... array of target specifier strings ... */],
+    issuedBefore: number (timestamp), /* (optional) a Unix timestamp where tokens issued before this time are invalidated */
+    allowReauthMargin: bool /* (optional) if true, allows a 30 seconds margin for clients to obtain a new auth token online, default false */
+}
+```
+
+The request body has the following properties:
+
+- targets := An array of "target specifier":#target-specifiers strings.
+- issuedBefore := If @issuedBefore@ is not specified, it is set to the current time. A request with an @issuedBefore@ in the future or more than an hour in the past will be rejected.
+- allowReauthMargin := The @allowReauthMargin@ bool permits a token renewal cycle to take place without needing established connections to be dropped, by postponing enforcement to 30 seconds in the future, and sending any existing connections a hint to obtain (and upgrade the connection to use) a new token. It defaults to @false@, meaning that the effect is near-immediate.
+
+When invoking the revocation API, a client must prove possession of the key used to issue the tokens to be revoked. You can't use a key to revoke a token issued from a different key. This can be done by using basic authentication, using the API key itself.
+
+There can be a maximum of 100 target specifiers in a single request.
+
+The token revocation API is rate-limited, so there is a maximum global aggregate rate of revocation requests per app. The rate is configurable by Ably at the application level. As part of the process of enabling revocation on an app Ably will ask you what the maximum rate of token revocations on that app needs to be, and then provision that capacity. The decision is not permanent, it can be changed by Ably on request at any time.
+
+Note that revocation is not a permanent prohibition of any client, connection, @clientId@ or channel. It invalidates token credentials issued before a certain point in time. Ultimately, the app that issues tokens is the arbiter of whether or not a client is able to continue to interact with the Ably service.
+
+Do not expect to be able to enable token revocation on an app, or the @revocableTokens@ attribute of a key, and then revoke a token that's been used since before you made that change. If you want to be able to revoke a token, revocation should have been enabled at the time that the token was issued and first used.
```

Comment:
I wonder if this paragraph should be under the "revocable tokens" header and a one-liner? E.g:

"Token revocation can only revoke tokens that were issued by an API key that had the @revocableTokens@ attribute set before the token was issued."

Reply:
Modified and moved.